rust/src/container.rs

```diff
@@ -387,6 +398,20 @@ pub fn container_encapsulate(args: Vec<String>) -> CxxResult<()> {
         })
         .collect::<Result<_>>()?;
 
+    let package_structure = match &opt.ocimeta_json {
+        Some(ocimeta_json) => {
+            let path = std::path::Path::new(ocimeta_json);
+            if path.is_file() {
+                let manifest = oci_spec::image::ImageManifest::from_file(path)
+                    .expect("parse ImageManifest from ocimeta");
```

Comment:
Let's use `?` to propagate errors here instead of crashing if the manifest is invalid.

Reply:
It has trouble converting a `OciSpecError` to `CxxError` as a result of the return type being `CxxResult`

Reply:
Ah yes.  Try `let parsed_manifest = opt.ocimeta_json.as_ref().map(|p| oci_spec::image::ImageManifest::from_file(&p).map_err(anyhow::Error::new)).transpose()?; or so

If you search for `.map_err(anyhow::Error::msg)` in the codebase you can see other places where we use this to convert error types.